deploy/deploy_marketplace.sh

```diff
@@ -2,195 +2,72 @@
 
 set -ex
 
-RED='\033[0;31m'
-NO_COLOR='\033[0m'
```

Comment:
this change is irrelevant. If you think that this should be dropped, please do it in a separate PR.

Reply:
Because when you have a cluster (clean one), you already have 3 catalogsources: redhat-operators, community, and certified-operators. So no need for preparation.
In case of QE/Dev, custom catalogsources are prepared earlier, and already used for other operators (not just CNV)

Should someone, needs a custom catsrc, then he can't use the "full" script as it's not supporting the creation of a plain catalogSource (ONLY by operatorSource, from QUAY).

IMO, an OLM catalogSource creation script, should be added to repo as an extra utility, and to support plain/opsrc catsrc creation.

Reply:
I don't like long scripts myself. If you want to split out some logic to another script, please do it. But please take care of existing use cases

Reply:
As mentioned before, for 99.99% of deploy_marketplace.sh, creation of catalogSOurce is not required.  For other cases, the split is in a new script (with old content) deploy_operatoSource.sh.

Reply:
Currently we already have deploy_imageregistry.sh (to consume the bundle image as a local registry) and deploy_marketplace.sh (to deploy pointing to a remote app-registry) and we want to unify them (to avoid having to duplicate each fix to both of them).
Introducing a third script is definitively not an option.

Reply:
deploy_imageregistry is creating catalogSource not an operatorSource from QUAY.
Also this one should be splitted..
Sure, it's a good idea to unify them (just for catsrc/opsrc creation)